### PR TITLE
Anca m/tema8

### PR DESCRIPTION
**Pull request by AncaPatricia**

### CODE REVIEW

**Review by curtnazarEliz — APPROVED** [review]

File: src/test/java/ancaMarian/tests/tema8/tests/DisabledDynamicButtonsPageTest.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        DisabledDynamicButtonsPage disabledDynamicButtonsPage = new DisabledDynamicButtonsPage(driver);
+        disabledDynamicButtonsPage.waitForStartButtonToBeClickable();
+        disabledDynamicButtonsPage.clickOnStartButton();
+        disabledDynamicButtonsPage.waitForOneButtonToBeClickable();
+        disabledDynamicButtonsPage.clickOnOneButton();
+        disabledDynamicButtonsPage.waitForWaitMessageToDisappear();
+        disabledDynamicButtonsPage.clickOnTwoButton();
+        disabledDynamicButtonsPage.waitForWaitMessageToDisappear();
+        disabledDynamicButtonsPage.clickOnThreeButton();
+        WebElement confirmationMessage = driver.findElement(By.id("buttonmessage"));
```
Comment (95teo) [inline src/test/java/ancaMarian/tests/tema8/tests/DisabledDynamicButtonsPageTest.java:25]: Move this element in page object class and create a method to get it's text to use it in the assert. Thanks!

File: src/test/java/ancaMarian/tests/tema8/tests/DynamicButtonsSimpleTest.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        DynamicButtonsSimplePage dynamicButtonsSimplePage = new DynamicButtonsSimplePage(driver);
+        dynamicButtonsSimplePage.waitForStartButtonToBeVisible();
+        dynamicButtonsSimplePage.clickOnStartButton();
+        dynamicButtonsSimplePage.waitForOneButtonToBeVisible();
+        dynamicButtonsSimplePage.clickOnOneButton();
+        dynamicButtonsSimplePage.waitForWaitMessageToDisappear();
+        dynamicButtonsSimplePage.clickOnTwoButton();
+        dynamicButtonsSimplePage.waitForWaitMessageToDisappear();
+        dynamicButtonsSimplePage.clickOnThreeButton();
+        WebElement confirmationMessage = driver.findElement(By.id("buttonmessage"));
```
Comment (95teo) [inline src/test/java/ancaMarian/tests/tema8/tests/DynamicButtonsSimpleTest.java:23]: Move this element in page object class and create a method to get it's text to use it in the assert. Thanks!

File: src/test/java/ancaMarian/tests/tema8/tests/DisabledDynamicButtonsPageTest.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        Assert.assertEquals(confirmationMessage.getText(), "All Buttons Clicked");
```
Comment (95teo) [inline src/test/java/ancaMarian/tests/tema8/tests/DisabledDynamicButtonsPageTest.java:26]: Please also double check this scenario as it seems that the assertion is failing ->

java.lang.AssertionError: 
Expected :All Buttons Clicked
Actual   :Click Buttons In Order
<Click to see difference>


	at org.testng.Assert.fail(Assert.java:110)
	at org.testng.Assert.failNotEquals(Assert.java:1577)
	at org.testng.Assert.assertEqualsImpl(Assert.java:149)
	at org.testng.Assert.assertEquals(Assert.java:131)
	at org.testng.Assert.assertEquals(Assert.java:655)
	at org.testng.Assert.assertEquals(Assert.java:665)
	at ancaMarian.tests.tema8.tests.DisabledDynamicButtonsPageTest.dynamicButtonsPageTest(DisabledDynamicButtonsPageTest.java:26)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:577)
	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
	at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:677)
	at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:221)
	at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
	at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:969)
	at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:194)
	at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
	at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.testng.TestRunner.privateRun(TestRunner.java:829)
	at org.testng.TestRunner.run(TestRunner.java:602)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:437)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:431)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:391)
	at org.testng.SuiteRunner.run(SuiteRunner.java:330)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1256)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1176)
	at org.testng.TestNG.runSuites(TestNG.java:1099)
	at org.testng.TestNG.run(TestNG.java:1067)
	at com.intellij.rt.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:66)
	at com.intellij.rt.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:109)

**Review by 95teo — CHANGES_REQUESTED** [review]

**Comment by AncaPatricia** [comment]
1) in clasa DynamicButtonsSimplePage am adăugat o metoda care ia textul mesajului de confirmare, apoi am folosit metoda în assertul din DynamicButtonsSimpleTest.
2) in clasa DisabledDynamicButtonsPage am adăugat o metoda care ia textul mesajului de confirmare
3)in clasa DisabledDynamicButtonsPage nu am reușit sa prind exact mesajul corect de confirmare fiindcă se tot schimba, așa ca am folosit o booleana care returnează true dacă a aparut mesajul corect de confirmare
4) la o ultima verificare a suitelor a picat un test din clasa RefreshPageTest unde am mărit timpul de așteptare de la 2 sec la 5 secunde pana da refresh la pagina

File: src/test/java/ancaMarian/tests/tema8/pageObjects/DisabledDynamicButtonsPage.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    public void waitForStartButtonToBeClickable() {
+        wait.until(ExpectedConditions.visibilityOf(startButton));
+    }
+
+    public void clickOnStartButton() {
+        startButton.click();
+    }
+
+    public void waitForOneButtonToBeClickable() {
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("button01")));
```
Comment (95teo) [inline src/test/java/ancaMarian/tests/tema8/pageObjects/DisabledDynamicButtonsPage.java:45]: Replace visibilityOfElementLocated with elementToBeClickable. The problem is with the expected condition you are using. The elements are already loaded in the page from the beginning so the are visible from the start. What you need here is to wait for them to be clickable as they are disabled until previous buttons are clicked.

File: src/test/java/ancaMarian/tests/tema8/tests/DisabledDynamicButtonsPageTest.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        DisabledDynamicButtonsPage disabledDynamicButtonsPage = new DisabledDynamicButtonsPage(driver);
+        disabledDynamicButtonsPage.waitForStartButtonToBeClickable();
+        disabledDynamicButtonsPage.clickOnStartButton();
+        disabledDynamicButtonsPage.waitForOneButtonToBeClickable();
+        disabledDynamicButtonsPage.clickOnOneButton();
+        disabledDynamicButtonsPage.waitForWaitMessageToDisappear();
+        disabledDynamicButtonsPage.clickOnTwoButton();
+        disabledDynamicButtonsPage.waitForWaitMessageToDisappear();
+        disabledDynamicButtonsPage.clickOnThreeButton();
+        if (disabledDynamicButtonsPage.waitForConfirmationMessageToAppear()) {
```
Comment (95teo) [inline src/test/java/ancaMarian/tests/tema8/tests/DisabledDynamicButtonsPageTest.java:25]: Replace this if with the assertion of the expected message after all buttons were clicked. Similar with the one from DynamicButtonsSimpleTest class. Thanks!

**Review by 95teo — CHANGES_REQUESTED** [review]

**Review by sorinfoca — APPROVED** [review]

**Review by 95teo — APPROVED** [review]

**Pull request merged**